logstash-core/src/main/java/org/logstash/plugins/AliasRegistry.java

```diff
@@ -37,13 +49,91 @@ PluginCoordinate withName(String name) {
         }
     }
 
+    private static class AliasYamlLoader {
+
+        private String extractedHash;
+        private String yaml;
+
+        @SuppressWarnings("unchecked")
+        private Map<PluginCoordinate, String> loadAliasesDefinitions() {
+            try {
+                parseYamlFile("org/logstash/plugins/plugin_aliases.yml");
+            } catch (IllegalArgumentException badSyntaxExcp) {
+                LOGGER.warn("Malformed yaml file", badSyntaxExcp);
+                return Collections.emptyMap();
+            }
+
+            // calculate the hash-256
+            final String sha256Hex = DigestUtils.sha256Hex(yaml);
+            if (!sha256Hex.equals(extractedHash)) {
+                LOGGER.warn("Bad checksum value, expected {} but found {}", sha256Hex, extractedHash);
```

Comment:
I thought that the Ruby part being a command line tool, so "run and exit" mode, is correct that it raises an error instead of print a log that the user could potentially ignore. 

On the other side the Java part is used in "always up process" mode and the aliases don't necessarily impact the user, as in the case the user doesn't configure any pipeline with any alias. 
If the user has a pipeline that doesn't start because of the alias can't be found, in that case Logstash stops and the user can find the reason in the log.

But I'm open to stop Logstash process on alias processing. LMKWDYT